spec/features/visitors/i18n_spec.rb

```diff
@@ -87,14 +87,13 @@
   end
 
   context 'visit homepage with host parameter' do
-    it 'does not include the host parameter in the language link URLs' do
+    it 'maintains query parameters as query parameters' do
       visit '/fr?host=test.com'
 
       %w[en es fr].each do |locale|
-        expect(page).to_not have_link(
```

Comment:
Problem with `to_not` is this wouldn't have matched even if there was a problematic link, since the links are hidden by default in the dropdown, hence the `visible: :all` and change to positive assertion with the changes here.